open_city_profile/tests/test_oidc_token_exchange.py

```diff
@@ -0,0 +1,31 @@
+from unittest.mock import MagicMock
+
+import pytest
+from oauthlib.oauth2 import OAuth2Error
+from requests_oauthlib import OAuth2Session
+
+from open_city_profile.exceptions import TokenExchangeError
+from open_city_profile.oidc import TunnistamoTokenExchange
+
+
+def test_authorization_code_exchange_successful(mocker):
+    tte = TunnistamoTokenExchange()
+    token_response = {"scope": "token"}
+    response = MagicMock()
+    response.json.return_value = token_response
+    mocker.patch.object(TunnistamoTokenExchange, "get_oidc_config")
+    mocker.patch.object(OAuth2Session, "fetch_token")
+    mocker.patch.object(OAuth2Session, "get", return_value=response)
```

Comment:
If I read this correctly, it mocks out quite a lot. If all network requests are done with `requests` (the library), there's also another option to only mock out the requests, like e.g. in [these tests](https://github.com/City-of-Helsinki/open-city-profile/blob/develop/open_city_profile/tests/test_graphql_authentication.py#L40-L45).

Reply:
Just realized that some tests use `requests-mock`, some use `responses`. The two libraries are probably solving the exact same problem and it could be possible to use only one of them. But no need to tackle that in this PR.

Reply:
I'll consider that alternative way of mocking the token fetching.